vllm/model_executor/model_loader/weight_utils.py

```diff
@@ -605,22 +607,23 @@ def safetensors_weights_iterator(
                 state_dict = load(f.read())
             yield from state_dict.items()
         elif safetensors_load_strategy == "torchao":
-            if not torchao_version_at_least("0.14.0"):
+            if not torchao_version_at_least("0.15.0"):
                 raise ValueError(
-                    "Please use torchao version >= 0.14.0 \
+                    "Please use torchao version >= 0.15.0 \
                         to load torchao safetensors checkpoint"
                 )
             from torchao.prototype.safetensors.safetensors_support import (
                 unflatten_tensor_state_dict,
             )
 
             with safe_open(st_file, framework="pt") as f:
-                state_dict = {}
                 for name in f.keys():  # noqa: SIM118
                     state_dict[name] = f.get_tensor(name)
                 metadata = f.metadata()
-                updated_state_dict = unflatten_tensor_state_dict(state_dict, metadata)
-            yield from updated_state_dict.items()
+                unflattened_state_dict, state_dict = unflatten_tensor_state_dict(
```

Comment:
we should add a comment to talk about what is happening here I think, around (1) sharded checkpoint and the issue for torchao tensor (2) the solution: the the unmatched tensors are left in state_dict and we expect it to be handled by next iteration in conjunction with other tensors that's loaded from the sharded checkpoint.